selftests/functional/test_utils.py

```diff
@@ -155,5 +160,35 @@ def test_process_run(self):
     def tearDown(self):
         shutil.rmtree(self.base_logdir)
 
+
+def file_lock_action(args):
+    path, players = args
+    max_individual_timeout = 1
+    max_timeout = max_individual_timeout * players
+    with FileLock(path, max_timeout) as lock:
+        sleeptime = random.random()
```

Comment:
You're basically proposing a more aggressive test.  I thought about having multiple versions, with different wait times and number of players.  The fact is that this setting caught bugs in the original implementation of FileLock, and seemed suitable for running on other people's machines - who may not use 486s anymore, but may not like a "fork bomb" like test, even when asking for longer tests.

In the end, I'd doubt any approach is wrong.  You suggestion can indeed be valid, but as we mentioned before, it ought to be given as that: a suggestion.